### PR TITLE
[android] Fixed "...more" button in bookmark's list description

### LINKED ISSUE #4813: [android] ...more button is not clickable in bookmark's list description
There are several options that can be done:
1. Expand TextView to see the full description while keeping it scrollable and other bookmarks below accessible.
2. Open the List Settings dialog where the full description is displayed and is editable.

Need to compare the pros and cons of both and choose the best approach.

Bonus question: can HTML be rendered in the bookmark's list description? A separate issue may be created for HTML rendering if necessary.

#4812 should be merged first to see the "...more" button.

### PR DESCRIPTION
**Pull request by ANS-UXI**
Fixes #4813
The "...more" button is now functional and is only visible for descriptions more than 2 lines.
ScreenRecord

### CODE REVIEW

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 13

Is it possible to increase size of text more/less ?

**Comment by ANS-UXI** [comment]
> white_check_mark Pixel 6 - Android 13
> 
> Is it possible to increase size of text more/less ?

Yes! Do you want me to increase its size?

**Comment by ANS-UXI** [comment]
| Before | After |
| --------- | ------ |
| Screenshot_2023-06-11-13-18-03-744_app organicmaps debug | Screenshot_2023-06-11-13-19-37-387_app organicmaps debug |

**Comment by Jean-BaptisteC** [comment]
Size is better :)

**Review by biodranik — COMMENTED** [review]
1. Is the whole text area including More button tappable?
2. Is it possible to increase a margin between text and More button?

**Comment by ANS-UXI** [comment]
> 1. Is the whole text area including More button tappable?
> 
>     2. Is it possible to increase a margin between text and More button?

**Comment by ANS-UXI** [comment]
Description is also now clickable and increased a margin of "2dp" between text and more button. Should I further increase the margin or it is fine?

video

**Review by biodranik — COMMENTED** [review]
1. Now, when the whole description area is clickable, there is no need to have a larger More button, it's size can be reduced.
2. Changing texts should be done using our strings.txt file. Let's not change more and let's not use Less at all. Users may scroll or tap again to hide the description.
3. After text is edited and saved, and returning back to the bookmark list, it's not immediately updated in the displayed description textview. That should be fixed too.
4. Pasting large piece of HTML text copied from the browser into the list description crashes it (but works on the master):
```
06-17 08:12:36.336  1128  1128 E AndroidRuntime: FATAL EXCEPTION: main
06-17 08:12:36.336  1128  1128 E AndroidRuntime: Process: app.organicmaps.beta, PID: 1128
06-17 08:12:36.336  1128  1128 E AndroidRuntime: java.lang.ArrayIndexOutOfBoundsException: length=31; index=-1
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.text.DynamicLayout.getBlockIndex(DynamicLayout.java:646)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.widget.Editor.drawHardwareAccelerated(Editor.java:1820)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.widget.Editor.onDraw(Editor.java:1789)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.widget.TextView.onDraw(TextView.java:7763)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20369)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19314)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20372)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at androidx.core.widget.NestedScrollView.draw(Unknown Source:0)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19314)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20372)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19314)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20372)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at com.android.internal.policy.DecorView.draw(DecorView.java:974)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19314)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:686)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:692)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:800)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl.draw(ViewRootImpl.java:3501)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:3288)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2823)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1785)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7833)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:723)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:658)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:789)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:98)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6942)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

**Review by rtsisyk — APPROVED** [review]

**Comment by ANS-UXI** [comment]
> 1. Now, when the whole description area is clickable, there is no need to have a larger More button, it's size can be reduced.
> 
>     2. Changing texts should be done using our strings.txt file. Let's not change more and let's not use Less at all. Users may scroll or tap again to hide the description.
> 
>     3. After text is edited and saved, and returning back to the bookmark list, it's not immediately updated in the displayed description textview. That should be fixed too.
> 
>     4. Pasting large piece of HTML text copied from the browser into the list description crashes it (but works on the master):
> 
> 
> ```
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: FATAL EXCEPTION: main
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: Process: app.organicmaps.beta, PID: 1128
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: java.lang.ArrayIndexOutOfBoundsException: length=31; index=-1
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.text.DynamicLayout.getBlockIndex(DynamicLayout.java:646)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.widget.Editor.drawHardwareAccelerated(Editor.java:1820)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.widget.Editor.onDraw(Editor.java:1789)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.widget.TextView.onDraw(TextView.java:7763)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20369)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19314)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20372)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at androidx.core.widget.NestedScrollView.draw(Unknown Source:0)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19314)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20372)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19314)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19305)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20092)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.drawChild(ViewGroup.java:4421)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:4207)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.draw(View.java:20372)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at com.android.internal.policy.DecorView.draw(DecorView.java:974)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.View.updateDisplayListIfDirty(View.java:19314)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:686)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:692)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:800)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl.draw(ViewRootImpl.java:3501)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:3288)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2823)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1785)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7833)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:723)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:658)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:789)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:98)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6942)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
> 06-17 08:12:36.336  1128  1128 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
> ```

1. Done.
2. Done.
3. #4826 , will start working on that.
4. Here is a very long HTML file that I used for testing.
[h.txt](https://github.com/organicmaps/organicmaps/files/11790192/h.txt)
On my emulator(Pixel 3a / Android 13) it crashes both on master and my branch. 
```
Transaction too large, intent: Intent { cmp=app.organicmaps.debug/app.organicmaps.bookmarks.BookmarkListActivity (has extras) }, extras size: 537748, icicle size: 0
2023-06-19 21:54:09.645   626-1261  BpBinder                system_process                       W  Large outgoing transaction of 541408 bytes, interface descriptor <uncached descriptor>, code 53
2023-06-19 21:54:09.645   626-1261  JavaBinder              system_process                       E  !!! FAILED BINDER TRANSACTION !!!  (parcel size = 541408)
2023-06-19 21:54:09.645   626-1261  ActivityTaskManager     system_process                       E  Second failure launching app.organicmaps.debug/app.organicmaps.bookmarks.BookmarkListActivity, giving up
                                                                                                    android.os.TransactionTooLargeException: data parcel size 541408 bytes
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:584)
                                                                                                    	at android.app.IApplicationThread$Stub$Proxy.scheduleTransaction(IApplicationThread.java:1900)
                                                                                                    	at android.app.servertransaction.ClientTransaction.schedule(ClientTransaction.java:136)
                                                                                                    	at com.android.server.wm.ClientLifecycleManager.scheduleTransaction(ClientLifecycleManager.java:47)
                                                                                                    	at com.android.server.wm.ActivityTaskSupervisor.realStartActivityLocked(ActivityTaskSupervisor.java:926)
                                                                                                    	at com.android.server.wm.RootWindowContainer$AttachApplicationHelper.test(RootWindowContainer.java:3607)
                                                                                                    	at com.android.server.wm.RootWindowContainer$AttachApplicationHelper.test(RootWindowContainer.java:3557)
                                                                                                    	at com.android.server.wm.ActivityRecord.forAllActivities(ActivityRecord.java:4563)
                                                                                                    	at com.android.server.wm.WindowContainer.forAllActivities(WindowContainer.java:1664)
                                                                                                    	at com.android.server.wm.WindowContainer.forAllActivities(WindowContainer.java:1658)
                                                                                                    	at com.android.server.wm.RootWindowContainer$AttachApplicationHelper.accept(RootWindowContainer.java:3595)
                                                                                                    	at com.android.server.wm.RootWindowContainer$AttachApplicationHelper.accept(RootWindowContainer.java:3557)
                                                                                                    	at com.android.server.wm.Task.forAllRootTasks(Task.java:3161)
                                                                                                    	at com.android.server.wm.WindowContainer.forAllRootTasks(WindowContainer.java:2045)
                                                                                                    	at com.android.server.wm.WindowContainer.forAllRootTasks(WindowContainer.java:2045)
                                                                                                    	at com.android.server.wm.WindowContainer.forAllRootTasks(WindowContainer.java:2045)
                                                                                                    	at com.android.server.wm.WindowContainer.forAllRootTasks(WindowContainer.java:2045)
                                                                                                    	at com.android.server.wm.WindowContainer.forAllRootTasks(WindowContainer.java:2045)
                                                                                                    	at com.android.server.wm.WindowContainer.forAllRootTasks(WindowContainer.java:2045)
                                                                                                    	at com.android.server.wm.WindowContainer.forAllRootTasks(WindowContainer.java:2038)
                                                                                                    	at com.android.server.wm.RootWindowContainer$AttachApplicationHelper.process(RootWindowContainer.java:3573)
                                                                                                    	at com.android.server.wm.RootWindowContainer.attachApplication(RootWindowContainer.java:1814)
                                                                                                    	at com.android.server.wm.ActivityTaskManagerService$LocalService.attachApplication(ActivityTaskManagerService.java:6239)
                                                                                                    	at com.android.server.am.ActivityManagerService.attachApplicationLocked(ActivityManagerService.java:4934)
                                                                                                    	at com.android.server.am.ActivityManagerService.attachApplication(ActivityManagerService.java:5024)
                                                                                                    	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:2339)
                                                                                                    	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2630)
                                                                                                    	at android.os.Binder.execTransactInternal(Binder.java:1280)
                                                                                                    	at android.os.Binder.execTransact(Binder.java:1244)
```
But on my physical device(Redmi 6 pro / Android 9) it definitely takes some time to load up but doesn't crashes both on master as well as on my branch.

File: android/src/app/organicmaps/bookmarks/BookmarkListAdapter.java [hunk 1]
```diff
@@ -513,4 +518,27 @@ public Object getItem(int position)
...
+        moreBtn.setVisibility(View.GONE);
+    });
+  }
+
+  private void onMoreButtonClicked(TextView moreBtn, TextView text)
+  {
+    if (text.getMaxLines() == 2)
+      text.setMaxLines(Integer.MAX_VALUE);
+    else
+      text.setMaxLines(2);
```
Comment (vng) [inline android/src/app/organicmaps/bookmarks/BookmarkListAdapter.java:542]: We have "2" four time in this module. Should take out as a separate constant inside Adapter class.

**Review by biodranik — APPROVED** [review]
LGTM if it works properly with different text and html sizes.

**Pull request merged**